### PR TITLE
Update Telemetry Tasks

### PR DESCRIPTION
**Pull request by gargnupur**
Updated Distributed Tracing Task with new screenshots

### CODE REVIEW

File: content/docs/tasks/telemetry/metrics-logs/index.md [hunk 1]
```diff
@@ -38,8 +38,9 @@ as the example application throughout this task.
     spec:
       value: "2" # count each request twice
       dimensions:
-        source: source.service | "unknown"
-        destination: destination.service | "unknown"
+        reporter: conditional((context.reporter.kind | "inbound") == "outbound", "client", "server")
+        source: source.name | "unknown"
```
Comment (douglas-reid) [inline content/docs/tasks/telemetry/metrics-logs/index.md]: should we use `source.workload.name` and `destination.workload.name` here instead?

or `source.labels["app"]` and `destination.labels["app"]` ?

same question below for the `newlog` changes.

with these changes, the data drops to be something like:
```
destination="details-v1",instance="172.17.0.12:42422",job="istio-mesh",message="twice the fun!",reporter="client",source="productpage-v1"
```

File: content/docs/tasks/telemetry/distributed-tracing/index.md [hunk 2]
```diff
@@ -61,11 +61,11 @@ Although every service has the same label, `istio-proxy`, because the tracing is
...
 
-The first line represents the external call to the `productpage` service. The label `192.168.64.3:32000` is the host
+The first line represents the external call to the `productpage` service. The label `192.168.39.95:31380` is the host
 value used for the external request (i.e., $GATEWAY_URL). As you can see in the trace,
-the request took a total of roughly 290ms to complete. During its execution, the `productpage` called the `details` service,
-which took about 24ms, and then called the `reviews` service.
-The `reviews` service took about 243ms to execute, including a 15ms call to `ratings`.
+the request took a total of roughly 45.56ms to complete. During its execution, the `productpage` called the `details` service,
+which took about 1.61ms, and then called the `reviews` service.
+The `reviews` service took about 36.44ms to execute, including a 1.1ms call to `ratings`.
```
Comment (douglas-reid) [inline content/docs/tasks/telemetry/distributed-tracing/index.md]: Do you think we should mention the client and server spans here?

File: content/docs/tasks/telemetry/metrics-logs/index.md (hunk 1 above)
Comment (gargnupur) [reply content/docs/tasks/telemetry/metrics-logs/index.md]: Thanks.. changed.

File: content/docs/tasks/telemetry/distributed-tracing/index.md (hunk 2 above)
Comment (gargnupur) [reply content/docs/tasks/telemetry/distributed-tracing/index.md]: Thanks..Fixed

**Comment by geeknoid** [comment]
Many linter errors need to be fixed in this PR before it can be merged.

**Comment by gargnupur** [comment]
@geeknoid : Can you please take a look again.. fixed the linter errors

**Comment by geeknoid** [comment]
@douglas-reid Please give this a once over.

File: content/docs/tasks/telemetry/servicegraph/index.md [hunk 3]
```diff
@@ -19,16 +19,8 @@ the example application throughout this task.
...
-    Servicegraph add-on.
-
-    In Kubernetes environments, execute the following command:
-
-    {{< text bash >}}
-    $ kubectl apply -f install/kubernetes/addons/servicegraph.yaml
-    {{< /text >}}
-
-1.  Verify that the service is running in your cluster.
+1.  Verify that the service is running in your cluster  (since 1.0 the
```
Comment (geeknoid) [inline content/docs/tasks/telemetry/servicegraph/index.md]: Remove the stuff in parenthesis

File: content/docs/tasks/telemetry/using-istio-dashboard/index.md [hunk 4]
```diff
@@ -16,21 +16,24 @@ the example application throughout this task.
 
 * [Install Istio](/docs/setup/) in your cluster and deploy an
   application.
-* [Install the Prometheus add-on](/docs/tasks/telemetry/querying-metrics/).
 
 ## Viewing the Istio Dashboard
 
-1.  To view Istio metrics in a graphical dashboard install the Grafana add-on.
+1.  Verify that the prometheus service is running in your cluster (since 0.8 the
```
Comment (geeknoid) [inline content/docs/tasks/telemetry/using-istio-dashboard/index.md]: Remove the parenthesis stuff

File: content/docs/tasks/telemetry/using-istio-dashboard/index.md [hunk 5]
```diff
@@ -16,21 +16,24 @@ the example application throughout this task.
...
+    $ kubectl -n istio-system get svc prometheus
+    NAME         CLUSTER-IP     EXTERNAL-IP   PORT(S)    AGE
+    prometheus   10.59.241.54   <none>        9090/TCP   2m
     {{< /text >}}
 
-1.  Verify that the service is running in your cluster.
+1.  Verify that the Grafana service is running in your cluster.
 
-    In Kubernetes environments, execute the following command:
+    In Kubernetes environments, execute the following command (since 1.0 the
```
Comment (geeknoid) [inline content/docs/tasks/telemetry/using-istio-dashboard/index.md]: Parens again

File: content/docs/tasks/telemetry/using-istio-dashboard/index.md [hunk 6]
```diff
@@ -70,43 +73,73 @@ the example application throughout this task.
...
     generated. It will look similar to:
 
-    {{< image width="100%" ratio="56.57%"
+    {{< image width="100%" ratio="54.51%"
     link="./dashboard-with-traffic.png"
     caption="Istio Dashboard With Traffic"
     >}}
 
-> `$GATEWAY_URL` is the value set in the [Bookinfo](/docs/examples/bookinfo/) example.
+    This gives the global view of Istio Mesh along with services and workloads in the mesh.
```
Comment (geeknoid) [inline content/docs/tasks/telemetry/using-istio-dashboard/index.md]: Istio mesh -> the mesh

File: content/docs/tasks/telemetry/using-istio-dashboard/index.md [hunk 7]
```diff
@@ -70,43 +73,73 @@ the example application throughout this task.
...
+    [http://localhost:3000/dashboard/db/istio-service-dashboard](http://localhost:3000/dashboard/db/istio-service-dashboard) in your web browser.
+
+    The Istio Service Dashboard will look similar to:
+
+    {{< image width="100%" ratio="51.65%"
+    link="./istio-service-dashboard.png"
+    caption="Istio Service Dashboard"
+    >}}
+
+    This gives details about metrics for the service and then client workloads(workloads that are calling this service)
```
Comment (geeknoid) [inline content/docs/tasks/telemetry/using-istio-dashboard/index.md]: Add a space before the two (

File: content/docs/tasks/telemetry/using-istio-dashboard/index.md [hunk 8]
```diff
@@ -70,43 +73,73 @@ the example application throughout this task.
...
 Dashboard is built to be used in conjunction with the default Istio metrics
 configuration and a Prometheus backend.
 
 The Istio Dashboard consists of three main sections:
-1. A Global Summary View. This section provides high-level summary of HTTP
-   requests flowing through the service mesh.
-1. A Mesh Summary View. This section provides slightly more detail than the
-   Global Summary View, allowing per-service filtering and selection.
+
+1. A Mesh Summary View. This section provides Global Summary view of the Istio Mesh and shows HTTP/gRPC and TCP
```
Comment (geeknoid) [inline content/docs/tasks/telemetry/using-istio-dashboard/index.md]: Istio mesh -> the mesh

File: content/docs/tasks/telemetry/using-istio-dashboard/index.md [hunk 9]
```diff
@@ -70,43 +73,73 @@ the example application throughout this task.
...
+   workloads in Istio Mesh.
```
Comment (geeknoid) [inline content/docs/tasks/telemetry/using-istio-dashboard/index.md]: Istio mesh -> the mesh

File: content/docs/tasks/telemetry/using-istio-dashboard/index.md [hunk 10]
```diff
@@ -70,43 +73,73 @@ the example application throughout this task.
...
+
 1. Individual Services View. This section provides metrics about requests and
-   responses for each individual service within the mesh (HTTP and TCP).
+   responses for each individual service within the mesh (HTTP/gRPC and TCP).
+   Also, give metrics about client and service workloads for this service.
+
+1. Individual Workloads View: This section provides metrics about requests and
+   responses for each individual workloads within the mesh (HTTP/gRPC and TCP).
+   Also, give metrics about inbound workloads and outbound services for this workload.
```
Comment (geeknoid) [inline content/docs/tasks/telemetry/using-istio-dashboard/index.md]: "inbound workloads" and "outbound services" doesn't quite work. Workloads and services aren't directional. Do you mean inbound and outbound traffic?

File: content/docs/tasks/telemetry/distributed-tracing/index.md [hunk 11]
```diff
@@ -61,11 +61,13 @@ Although every service has the same label, `istio-proxy`, because the tracing is
...
-The first line represents the external call to the `productpage` service. The label `192.168.64.3:32000` is the host
+The first line represents the external call to the `productpage` service. The label `192.168.39.119:31380` is the host
 value used for the external request (i.e., $GATEWAY_URL). As you can see in the trace,
-the request took a total of roughly 290ms to complete. During its execution, the `productpage` called the `details` service,
-which took about 24ms, and then called the `reviews` service.
-The `reviews` service took about 243ms to execute, including a 15ms call to `ratings`.
+the request took a total of roughly 34.39ms to complete. During its execution, the `productpage` called the `details` service,
+which took about 1.72ms, and then called the `reviews` service.
+The `reviews` service took about 22.99ms to execute. The first `reviews` span taking 1.6ms represent server span wherein
+it's receiving call from `productpage`. Second `reviews` span taking 9.95 ms represent client span wherein it's calling
```
[the remark this answers is not in the document]
Comment (gargnupur) [reply content/docs/tasks/telemetry/distributed-tracing/index.md]: I am trying to differentiate between client and server role of reviews here..

File: content/docs/tasks/telemetry/querying-metrics/index.md [hunk 12]
```diff
@@ -19,7 +19,7 @@ application.
 ## Querying Istio Metrics
 
 1.  Verify that the prometheus service is running in your cluster (since 0.8 the
```
[the remark this answers is not in the document]
Comment (gargnupur) [reply content/docs/tasks/telemetry/querying-metrics/index.md]: Fixed

File: content/docs/tasks/telemetry/servicegraph/index.md (hunk 3 above)
Comment (gargnupur) [reply content/docs/tasks/telemetry/servicegraph/index.md]: Fixed

File: content/docs/tasks/telemetry/using-istio-dashboard/index.md (hunk 4 above)
Comment (gargnupur) [reply content/docs/tasks/telemetry/using-istio-dashboard/index.md]: Fixed

File: content/docs/tasks/telemetry/using-istio-dashboard/index.md (hunk 5 above)
Comment (gargnupur) [reply content/docs/tasks/telemetry/using-istio-dashboard/index.md]: Fixed

File: content/docs/tasks/telemetry/using-istio-dashboard/index.md (hunk 6 above)
Comment (gargnupur) [reply content/docs/tasks/telemetry/using-istio-dashboard/index.md]: Fixed

File: content/docs/tasks/telemetry/using-istio-dashboard/index.md (hunk 7 above)
Comment (gargnupur) [reply content/docs/tasks/telemetry/using-istio-dashboard/index.md]: Fixed

File: content/docs/tasks/telemetry/using-istio-dashboard/index.md (hunk 8 above)
Comment (gargnupur) [reply content/docs/tasks/telemetry/using-istio-dashboard/index.md]: Fixed

File: content/docs/tasks/telemetry/using-istio-dashboard/index.md (hunk 9 above)
Comment (gargnupur) [reply content/docs/tasks/telemetry/using-istio-dashboard/index.md]: Fixed

**Comment by gargnupur** [comment]
Circleci failure is about an external link not reachable..

**Comment by geeknoid** [comment]
The Ci failures are unrelated to this PR and have been fixed.

File: content/docs/tasks/telemetry/distributed-tracing/index.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (douglas-reid) [reply content/docs/tasks/telemetry/distributed-tracing/index.md]: I think this needs to be rewritten with something like:

> The `productpage` to `reviews` call is represented by two spans in the trace. The first of the two spans (labeled *productpage* reviews.default.svc.cluster.local:9080/*) represents the client-side span for the call. It took `24.13ms`. The second span (labeled *reviews* reviews.default.svc.cluster.local:9080/*) is a child of the first span and represents the server-side span for the call. It took `22.99ms`.

> The trace for the call to the `reviews` services reveals two subsequent RPCs in the trace. The first is to the `istio-policy` service, reflecting the server-side `Check` call made for the service to authorize access. The second is the call out to the `ratings` service.

[the remark this answers is not in the document]
Comment (gargnupur) [reply content/docs/tasks/telemetry/distributed-tracing/index.md]: Thanks.. changed

File: content/docs/tasks/telemetry/using-istio-dashboard/index.md (hunk 10 above)
Comment (gargnupur) [reply content/docs/tasks/telemetry/using-istio-dashboard/index.md]: Added above explanation in Visualize Workload Dashboards section

**Comment by douglas-reid** [comment]
Ready to merge? There are several issues being opened about items resolved in this PR. It'd be nice to have this in.

**Comment by gargnupur** [comment]
@douglas-reid : Yes, circleci failures are not related to this PR but some external site not being reachable.

**Comment by andraxylia** [comment]
please merge to the latest, I fixed those errors with a PR yesterday. If you see similar, it means the links are broken and you need to update them.

**Comment by gargnupur** [comment]
@andraxylia : Merged to latest and these are the 4 new links failing now:
https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/mongo  https://archive.istio.io/v0.7/docs/reference/config/istio.routing.v1alpha1/  https://www.envoyproxy.io/docs/envoy/latest/api-v1/cluster_manager/sds  https://docs.helm.sh/using_helm/ 
They are all valid links and looks like test server is not able to curl them

**Comment by andraxylia** [comment]
Those links are indeed valid. I rebuilt the circleci, if it keeps failing , you need an admin to look into this.

**Comment by gargnupur** [comment]
@andraxylia : Thanks, circleci is passing now!
@douglas-reid @geeknoid : Can you please review and approve PR ?

**Comment by gargnupur** [comment]
@mandarjog : Can you please help merge this PR?

**Comment by andraxylia** [comment]
Please lgtm:
@geeknoid
 @linsun
 @frankbu

**Pull request merged**